page_analyzer/utils/url.py

```diff
@@ -18,3 +19,25 @@ def normalize(url_name: str) -> str:
         port = ":" + str(p.port)
         normalized = normalized.replace(port, "")
     return normalized
+
+
+def get_url_status_code(url: str) -> int | None:
+    try:
+        r = requests.get(url)
+        r.raise_for_status()
+    except requests.exceptions.HTTPError as e:
+        print("Http error: ", e)
+        return None
+    except requests.exceptions.ConnectionError as e:
+        print("Connection error: ", e)
+        return None
+    except requests.exceptions.Timeout as e:
+        print("Timeout error:", e)
+        return None
+    except requests.exceptions.TooManyRedirects as e:
+        print("Too many redirects error:", e)
+        return None
+    except requests.exceptions.RequestException as e:
+        print("Some other request error", e)
+        return None
+    return r.status_code
```

Comment:
wow, that's impressive!

from my point of view, it makes sense to handle different error types only if you are going to act differently depending on the error type

as all the error types are handled in the same way (print + return None), I'd only keep the last `requests.exceptions.RequestException` except

feel free to ignore tho :^)